File: Counting Sort/CountingSort.java
public class CountingSort{
    public static int[] countingSort(int[] array){
        int max = 0;
        for (int element : array) {
            if (element > max)
                max = element;
        }
        int [] elementCount = new int[++max];
        for (int element : array)
            elementCount[element]++;
        int [] sorted_array = new int[array.length];
        int position = 0;
        for (int i = 0; i < max; i++) {
            for (int j = 0; j < elementCount[i]; j++)
                sorted_array[position++] = i;
        }
        return sorted_array;
    }

    public static void main(String[] args) {
        int[] pi = {3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5};
        int[] sortedPi = countingSort(pi);
        System.out.println("pi = " + java.util.Arrays.toString(pi));
        System.out.println("sortedPi = " + java.util.Arrays.toString(sortedPi));
    }
}

File: Counting Sort/counting_sort.js
function countingSort(unsortedArray) {
    let max = 0;
    for (let i = 0; i < unsortedArray.length; i++) {
        if (unsortedArray[i] > max)
            max = unsortedArray[i];
    }
    let elementCount = [];
    for (let i = 0; i < unsortedArray.length; i++) {
        if (elementCount[unsortedArray[i]])
            elementCount[unsortedArray[i]]++;
        else
            elementCount[unsortedArray[i]] = 1;
    }
    let sortedArray = [];
    for (let i = 0; i < elementCount.length; i++) {
        for (let j = 0; j < elementCount[i]; j++) {
            sortedArray.push(i);
        }
    }
    return sortedArray;
}